manuscripts/report.py

```diff
@@ -675,6 +676,22 @@ def period_name(cls, pdate, interval='quarter', offset=None):
 
         return name
 
+    def find_and_replace(self, directory, to_replace, replacement, file_type=None):
+        """
+        Looks for a string 'to_replace' in the 'directory' which has files containing
+        that string and replaces the string with its 'replacement' string. The type of
+        file to look for can be specified using 'file_type'.
+        """
+        if not file_type:
+            file_type = "**/*.tex"
+
+        for filepath in glob.iglob(os.path.join(os.path.abspath(directory), file_type), recursive=True):
```

Comment:
This is risky, because if some user wrongly specifies the directory to produce the report, this can potentially run through all a file tree, trying to change all files in it. Imagine for example that they say `/` is the directory...

I think it would be better to structure the code as:

* A function `replace_text` that looks for a string and substitutes it for some other. This could be a static method.
* A function `replace_text_dir` that runs replace_tex for all files in a certain directory with a certain extension, not recursive. The directory would be obtained from `self.data_dir`, so no need to include it as a parameter.

Do you find this reasonable?